### PR TITLE
Integration testing: cluster tx redirection

### PR DESCRIPTION
**Pull request by tohar-ayash**
Signed-off-by: tohar-ayash <toharayash1@gmail.com>

### CODE REVIEW

File: test/cluster/cluster_redirection_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	require.NoError(t, c.ShutdownServer(c.Servers[0]))
+	require.NoError(t, c.ShutdownServer(c.Servers[1]))
+
+	// only node 3 is alive => no quorum
+	require.Eventually(t, func() bool {
+		_, _, err = c.Servers[2].WriteDataTx(t, worldstate.DefaultDBName, "new-key-2", []byte("new-data-2"))
+		return err.Error() == "failed to submit transaction, server returned: status: 503 Service Unavailable, message: Cluster leader unavailable"
+	}, 60*time.Second, 100*time.Millisecond)
```
Comment (tock-ibm) [inline test/cluster/cluster_redirection_test.go:81]: This part is not needed in this test

File: test/cluster/cluster_redirection_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		require.Eventually(t, func() bool {
+			dataEnv, err := c.Servers[i].QueryData(t, worldstate.DefaultDBName, "key-"+strconv.Itoa(i), "admin")
+			if dataEnv != nil && dataEnv.GetResponse().GetValue() != nil {
+				dataVal := dataEnv.GetResponse().GetValue()
+				require.Equal(t, dataVal, []byte("data-"+strconv.Itoa(i)))
+				t.Logf("data: %+v", dataVal)
+			}
+			return dataEnv != nil && dataEnv.GetResponse().GetValue() != nil && err == nil
+		}, 30*time.Second, 100*time.Millisecond)
+	}
```
Comment (tock-ibm) [inline test/cluster/cluster_redirection_test.go:72]: here you should check that all servers have all 3 values

File: test/cluster/cluster_redirection_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	setupConfig := &setup.Config{
+		NumberOfServers:     3,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return nil
+		},
```
[the remark this answers is not in the document]
Comment (tohar-ayash) [reply test/cluster/cluster_redirection_test.go:38]: Done

File: test/cluster/cluster_redirection_test.go (hunk 1 above)
Comment (tohar-ayash) [reply test/cluster/cluster_redirection_test.go:81]: Done

File: test/cluster/cluster_redirection_test.go (hunk 2 above)
Comment (tohar-ayash) [reply test/cluster/cluster_redirection_test.go:72]: done

**Review by tock-ibm — DISMISSED** [review]

File: test/cluster/cluster_redirection_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import (
+	"github.com/hyperledger-labs/orion-server/internal/worldstate"
+	"io/ioutil"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hyperledger-labs/orion-server/pkg/types"
+	"github.com/hyperledger-labs/orion-server/test/setup"
+	"github.com/stretchr/testify/require"
```
Comment (tock-ibm) [inline test/cluster/cluster_redirection_test.go:15]: goimports

**Pull request merged**